File: Geeks for Geeks/YT Video Series wise/Arrays/Array-rotation/Java/ArrayRotationFirstApproach.java
public class ArrayRotationFirstApproach {

    public static void main(String[] args) {
        int[] intArray = {1,2,3,4,5,6};
        int numOfRotation = 2;
        printArray(rotateUptoNumOfRotation(intArray, numOfRotation));

    }

    // create a method to call : number of array rotation should be executed
    protected static int[] rotateUptoNumOfRotation(int[] intArray,int numOfRotation){
        for (int i = 0; i < numOfRotation; i++) {
           intArray = rotateOnce(intArray);
        }
        return intArray;
    }
    // Method : create one roation
    protected static int[] rotateOnce(int[] array){
        int temp = array[0];
        int i;
        for (i = 0; i < array.length - 1; i++) {
            array[i] = array[i+1];
        }
        array[i] = temp;
        return array;
    }
    // print the result
    protected static void printArray(int[] intArray){
        for (int i = 0; i < intArray.length; i++) {
            System.out.print(intArray[i]+" ");
        }
    }
}

File: Geeks for Geeks/YT Video Series wise/Arrays/Array-rotation/JavaScript/ArrayRotationFirstApproach.js


// create a function upto number of ratation number
function rotateUptoNumOfRotation(array,numOfRotation) {
  while(numOfRotation>0) {
    array.push(array.shift());
    numOfRotation--;
  } 
  return array;
}

// print the array
function printArray(array) {
 for (let i = 0; i < array.length; i++) {
   console.log(" "+array[i]);
 }
}

var array = [1, 2, 3, 4, 5, 6, 7];
//   console.log(array);
var numOfRoation = 2;
printArray(rotateUptoNumOfRotation(array, numOfRoation));
